spec/controllers/api/v1/punches_controller_spec.rb

```diff
@@ -0,0 +1,155 @@
+require 'spec_helper'
+
+describe Api::V1::PunchesController, :type => :controller do
+  let(:company) { create(:company) }
+  let(:user) { create(:user, :with_token, company: company) }
+  let(:project) { create(:project, company: user.company) }
+  let(:headers) { { token: user.token } }
+
+  describe 'POST api/v1/punches' do
+    context 'when api token is valid and params are valid' do
+      let(:params) {
+        {
+          "_json": [
+            {
+              "from": "2022-04-19T12:00:00.000Z",
+              "to": "2022-04-19T15:00:00.000Z",
+              "project_id": "#{project.id}"
+            },
+            {
+              "from": "2022-04-19T16:00:00.000Z",
+              "to": "2022-04-19T21:00:00.000Z",
+              "project_id": "#{project.id}"
+            }
+          ]
+        }
+      }
+
+      subject { post :create, params: params }
+
+      before do
+        request.headers.merge(headers)
+      end
+
+      it { is_expected.to have_http_status(:created) }
+
+      it 'returns content type json' do
+        expect(subject.content_type).to eq 'application/json; charset=utf-8'
+      end
+
+      it 'returns created punches' do
+        created_punches = JSON.parse(subject.body).map { |punch| {from: punch['from'], to: punch['to'], project_id: punch['project_id'].to_s } }
```

Comment:
You shouldn't need to remap everything :thinking: 
Does this work?
```
expect(response.body).to match_array(params[:_json])
```

Reply:
The response.body has more properties than the params that I send